core/src/SYCL/Kokkos_SYCL_Parallel_Scan.hpp

```diff
@@ -257,8 +257,11 @@ class ParallelScanSYCLBase {
     // Write results to global memory
     auto update_global_results = q.submit([&](sycl::handler& cgh) {
       auto global_mem                   = m_scratch_space;
-      auto result_ptr                   = m_result_ptr;
       auto result_ptr_device_accessible = m_result_ptr_device_accessible;
+      // The compiler failed with CL_INVALID_ARG_VALUE if using m_result_ptr
+      // directly.
+      auto device_accessible_result_ptr =
+          m_result_ptr_device_accessible ? m_result_ptr : nullptr;
```

Comment:
Please don't.  I'd rather have it called `result_ptr` than have both `result_ptr_device_accessible` and `device_accessible_result_ptr`.